starboard/shared/starboard/player/filter/testing/test_util.cc

```diff
@@ -181,14 +181,32 @@ std::vector<VideoTestParam> GetSupportedVideoTests() {
       const auto& video_stream_info = dmp_reader.video_stream_info();
       const std::string video_mime = dmp_reader.video_mime_type();
       const MimeType video_mime_type(video_mime.c_str());
-      if (SbMediaIsVideoSupported(
-              dmp_reader.video_codec(),
-              video_mime.size() > 0 ? &video_mime_type : nullptr, -1, -1, 8,
-              kSbMediaPrimaryIdUnspecified, kSbMediaTransferIdUnspecified,
-              kSbMediaMatrixIdUnspecified, video_stream_info.frame_width,
-              video_stream_info.frame_height, dmp_reader.video_bitrate(),
-              dmp_reader.video_fps(), false)) {
-        test_params.push_back(std::make_tuple(filename, output_mode));
+      // SbMediaIsVideoSupported may return false for gpu based decoder that in
+      // fact supports av1 or/and vp9 because the system can make async
+      // initialization at startup.
+      // To minimize probability of false negative we check result few times
+      static bool decoder_was_checked = false;
```

Comment:
Is it intentional that this is static, e.g. on next call it will default to whatever the last call resulted ? This seems a bit counterintuitive as the IsSupported call takes a lot of arguments that could be different on the next call ?

Reply:
There may be two cases. 
1) SbMediaIsVideoSupported returns false because of resources acquiring is still in progress. 
2) Decoder really isn't supported. 
I never faced with delays more then half a second so I supposed that 5 seconds were quite enough to ensure that the problem is not in resources acquiring. So I added static variable to escape the next waiting. But if you say that delay may be more than 5 sec I will delete this condition

Reply:
No need to change the logic, i think your explanation makes sense. Just for extra clarity I'd maybe consider changing the name of the variable to `decoder_has_been_checked_once`

Reply:
A different option could be to have `decoder_was_checked` as a global in an anonymous namespace.

Related to my above comment, this could have all platforms check `SbMediaIsVideoSupported()` up to 5 times if they really don't support the codec. This wouldn't be great especially for platforms that have expensive calls, like Android.

Reply:
I guess that more natural will be to add new variable to SbMediaIsVideoSupported interface. For instance "bool wait_result = false". In this case different callers (tests, player) could use needed option. And "internals" of implementation will be hidden inside internal section. But this way will cause much more code changes in both internal and shared sections. So I choose the simplest way. I would like to know your opinion about this way

Reply:
May be better to keep it as is to keep the surface of the change low - I think changing the `SbMediaIsVideoSupported` interface would be difficult, especially for a narrow use case. A different option may be to define a new configuration constant, `kMustWaitForGpuDecoder` or something, and use that to have only Xbox use this new implementation. At that point the change gets very clunky though; if performance is acceptable on Android I'd say we'd keep it like this.

Reply:
Let's take a look on this issue and suggested decision. 
1) It effects just tests, not player
2) Each test calls SbMediaIsVideoSupported a small amount of times (in our case 4 times)
3) The only one test can use multipled call (max 5 times separated by delay of 1 sec)
So if to launch player_filter_tests for some platform then in the worst scenario we will get 4 extra call of SbMediaIsVideoSupported and delay up to 5 sec. Is it so big price keeping in mind that the whole test takes at least few minutes?

Reply:
This change in isolation is fine. However, at least how I see it, we want to minimize overhead for every test change. A couple seconds is small compared to the run time of the test suite, but many such changes add up. In this case though, I don't see a better option.